torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -101,6 +102,9 @@ class AttributePropagator {
       ClearProfilingInformation(subgraph);
     };
     auto applyOptimizations = [](std::shared_ptr<Graph>& subgraph) {
+#ifndef C10_MOBILE
+      Autocast(subgraph);
```

Comment:
cc @dhruvbird do you know why mobile is including freezing ? I would think that freezing happens on the server first before mobile build

Reply:
The [optimize_for_mobile API](https://pytorch.org/docs/stable/mobile_optimizer.html) basically does a freeze pass since the lite-interpreter doesn't (or didn't till very recently?) support the `CALL` instruction, so we had to inline everything by freezing. Also I think some optimization passes will work better with a frozen module since it could lead to better operator fusion.

cc @iseeyuan @dreiss to confirm.

Reply:
I'm not sure.  optimize_for_mobile runs on the server build.  Our internal production and external recommended mobile settings don't include the JIT at all.

Reply:
ah! @dreiss good catch - yes the freeze pass does happen on the server build (and not the mobile build), and `C10_MOBILE` is defined only for the mobile build (and not the server build).

Reply:
In either case, this doesn't seem wrong, since this code will only be used when `C10_MOBILE` is not defined, but it does look confusing.

Reply:
@dhruvbird do you mind submitting a PR to clean it up?